### PR TITLE
chore: add github action to verify code

### LINKED ISSUE #253: enable verify in makefile and our github workflow
We actually have many scripts to verify/test our code but not used yet which is really  great loss here.

The very simple way to enable `verify` is to run `hack/verify-all.sh` but this needs to be verified.
```makefile
.PHONY: verify
verify:
	@hack/verify-all.sh
```

Besides, some scripts should not be used or optimzied. Overrall, this will be a `difficulty-medium` task. 

To enable this in our github workflow, we need to add another action or task.

### PR DESCRIPTION
**Pull request by zqq454224016**
#### What type of PR is this?
/kind chore

#### Which issue(s) this PR fixes

Fixes #253

### CODE REVIEW

**Comment by zqq454224016** [comment]
@Abirdcfly @0xff-dev @bjwswang

**Comment by 0xff-dev** [comment]
Delete the vendor directory.

File: hack/verify-all.sh [hunk 1]
```diff
@@ -10,9 +10,9 @@ source "${ROOT_PATH}/hack/lib/init.sh"
 go::verify_go_version
 
 bash "$ROOT_PATH/hack/verify-shfmt.sh"
-bash "$ROOT_PATH/hack/verify-vendor.sh"
+#bash "$ROOT_PATH/hack/verify-vendor.sh"
 bash "$ROOT_PATH/hack/verify-codegen.sh"
 bash "$ROOT_PATH/hack/verify-crdgen.sh"
-bash "$ROOT_PATH/hack/verify-golangci-lint.sh"
+bash "$ROOT_PATH/hack/verify-golangci-lint.sh" # mirror
```
Comment (zqq454224016) [inline hack/verify-all.sh:16]: According to .golanci.yaml , i dont find mirror linter added by kubebb.

Comment (zqq454224016) [inline hack/verify-all.sh:16]: no chart.yaml in this path

File: hack/verify-all.sh [hunk 2]
```diff
@@ -10,9 +10,9 @@ source "${ROOT_PATH}/hack/lib/init.sh"
 go::verify_go_version
 
 bash "$ROOT_PATH/hack/verify-shfmt.sh"
-bash "$ROOT_PATH/hack/verify-vendor.sh"
+#bash "$ROOT_PATH/hack/verify-vendor.sh"
 bash "$ROOT_PATH/hack/verify-codegen.sh"
 bash "$ROOT_PATH/hack/verify-crdgen.sh"
```
Comment (zqq454224016) [inline hack/verify-all.sh:15]: **make manifests** has same function.

File: hack/verify-all.sh [hunk 3]
```diff
@@ -10,9 +10,9 @@ source "${ROOT_PATH}/hack/lib/init.sh"
 go::verify_go_version
 
 bash "$ROOT_PATH/hack/verify-shfmt.sh"
-bash "$ROOT_PATH/hack/verify-vendor.sh"
+#bash "$ROOT_PATH/hack/verify-vendor.sh"
```
Comment (Abirdcfly) [inline hack/verify-all.sh:13]: The files here is from the previous general specification.

According to the current project, we can make some changes:

1. The vendor-related scripts are not needed and can be removed.
2. The helm-related scripts are not needed and can be removed.
3. The codegen-related scripts are not needed and can be removed. （ We are using `make generate`, and we may need to confirm the operation of this command.)
4. The crdgen-related scripts are not needed and can be removed. （ We are using `make manifests`, and we may need to confirm the operation of this command.)

File: hack/verify-crdgen.sh
[the remark this answers is not in the document]
Comment (Abirdcfly) [reply hack/verify-crdgen.sh:1]: https://github.com/git/git/blob/f9972720e9a405e4f6924a7cde0ed5880687f4d0/Documentation/RelNotes/2.30.0.txt#L46 This parameter was added from git version 2.30.0, 3 years ago.

File: Makefile [hunk 4]
```diff
@@ -260,3 +260,6 @@ catalog-build: opm ## Build a catalog image.
 catalog-push: ## Push a catalog image.
 	$(MAKE) docker-push IMG=$(CATALOG_IMG)
 
+.PHONY: verify
```
Comment (Abirdcfly) [inline Makefile:263]: The changes here are fine for me, with or without adding.

File: .golangci.yml [hunk 5]
```diff
@@ -34,7 +34,7 @@ linters:
     - govet
     - ineffassign
 #     - logcheck  # we do not need this linter
-    - mirror # add by kubebb
+#    - mirror # add by kubebb
```
Comment (Abirdcfly) [inline .golangci.yml:37]: https://github.com/kubebb/core/pull/269#discussion_r1299346853

File: hack/verify-crdgen.sh
[the remark this answers is not in the document]
Comment (zqq454224016) [reply hack/verify-crdgen.sh:1]: fine, my git version is 2.25.0.

File: hack/verify-all.sh (hunk 1 above)
Comment (zqq454224016) [reply hack/verify-all.sh:16]: It download 1.52.2 by default.

**Review by bjwswang — APPROVED** [review]

**Pull request closed without merging**